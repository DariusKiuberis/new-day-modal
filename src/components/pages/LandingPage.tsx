import { FC, useState } from 'react'
import { Button, Modal } from '@mantine/core';
import { useStyles } from './styles'
import Form from '../organisms/Form/Form';

interface Props { }

const LandingPage: FC<Props> = ({ }) => {
    const { classes } = useStyles();

    const [opened, setOpened] = useState(false);

    return (

        <div
            className={classes.container}
            data-testid='landing-page'
        >
            <Modal
                classNames={{ modal: classes.modal }}
                withCloseButton={true}
                centered
                opened={opened}
                onClose={() => setOpened(false)}
                size="70%"
                transition="fade"
                transitionDuration={600}
                transitionTimingFunction="ease"
            >
                <Form />
            </Modal>
            <Button
                data-testid='open-btn'
                className={classes.openButton}
                onClick={() => setOpened(true)}
                variant='default'
            >
                Open modal
            </Button>
        </div >
    )
}

export default LandingPage

