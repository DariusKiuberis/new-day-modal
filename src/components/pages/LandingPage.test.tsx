import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import LandingPage from './LandingPage';

// ignore code below, this is just to make the test work ( long story, but it's all about MantineProvider)
Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: vitest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: vitest.fn(), // deprecated
    removeListener: vitest.fn(), // deprecated
    addEventListener: vitest.fn(),
    removeEventListener: vitest.fn(),
    dispatchEvent: vitest.fn(),
  })),
});

function setup(tsx: any) {
  return {
    user: userEvent.setup(),
    ...render(tsx),
  }
}

describe('LandinPage', () => {
  it('renders LandingPage component', () => {
    render(<LandingPage />);
    expect(screen.getByTestId('landing-page')).toMatchSnapshot()
  });

  it('open button exists', () => {
    render(<LandingPage />);

    expect(screen.getByText(/Open/)).toBeInTheDocument();
  });

  it('by clicking Open button Modal component is visible', async () => {

    const { user } = setup(<LandingPage />)

    await user.click(screen.getByRole('button', { name: /open/i }))

    waitFor(() => expect(screen.getByTestId('close-btn')).toBeInTheDocument()
    )
    // screen.debug();
  });


});