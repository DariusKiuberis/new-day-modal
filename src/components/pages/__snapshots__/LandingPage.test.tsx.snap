// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LandinPage > renders LandingPage component 1`] = `
<div
  class="mantine-izct5y"
  data-testid="landing-page"
>
  <button
    class="mantine-UnstyledButton-root mantine-Button-root mantine-1r0y3pt"
    data-button="true"
    data-testid="open-btn"
    type="button"
  >
    <div
      class="mantine-3xbgk5 mantine-Button-inner"
    >
      <span
        class="mantine-qo1k2 mantine-Button-label"
      >
        Open modal
      </span>
    </div>
  </button>
</div>
`;
